### PR TITLE
fix: [CDS-47298]: nexus2 related fixes -> alignment and repository api

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
fix for ->
https://harness.atlassian.net/browse/CDS-47298
https://harness.atlassian.net/browse/CDS-47281
https://harness.atlassian.net/browse/CDS-47278
https://harness.atlassian.net/browse/CDS-47277

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx [hunk 1]
```diff
@@ -274,6 +322,19 @@ export function Nexus2Artifact({
...
+        text={
+          <Layout.Horizontal spacing="small">
+            <Text>{item.label}</Text>
+          </Layout.Horizontal>
+        }
+        disabled={fetchingRepository}
+        onClick={handleClick}
+      />
+    </div>
+  ))
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx:337]: `memoize` will be called on every render and a new memoized `itemRenderer` will be created. `useMemo` is more suitable here.

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx [hunk 2]
```diff
@@ -113,6 +120,47 @@ export function Nexus2Artifact({
...
+  const selectRepositoryItems = useMemo(() => {
+    return repositoryDetails?.data?.map(repository => ({
+      value: defaultTo(repository.repositoryName, ''),
+      label: defaultTo(repository.repositoryName, '')
+    }))
+  }, [repositoryDetails?.data])
+
+  const getRepository = (): { label: string; value: string }[] => {
+    if (fetchingRepository) {
+      return [{ label: 'Loading Repository...', value: 'Loading Repository...' }]
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx:159]: add label to strings

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx [hunk 3]
```diff
@@ -112,6 +123,53 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
...
+  const selectRepositoryItems = useMemo(() => {
+    return repositoryDetails?.data?.map(repository => ({
+      value: defaultTo(repository.repositoryName, ''),
+      label: defaultTo(repository.repositoryName, '')
+    }))
+  }, [repositoryDetails?.data])
+
+  const getRepository = (): { label: string; value: string }[] => {
+    if (fetchingRepository) {
+      return [{ label: 'Loading Repository...', value: 'Loading Repository...' }]
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:168]: add label to strings

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx [hunk 4]
```diff
@@ -212,6 +270,20 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
     return false
   }
 
+  const itemRenderer = memoize((item: { label: string }, { handleClick }) => (
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:273]: use `useMemo`

Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:273]: done

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx (hunk 3 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:168]: done

**Review by Eric761 — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx:337]: this needs to be done across all artifacts, will create a separate PR for it

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/Nexus2Artifact/Nexus2Artifact.tsx]: done

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx (hunk 4 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/Nexus2ArtifactSource/Nexus2ArtifactSource.tsx:278]: this needs to be done across all artifacts, will create a separate PR for it

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

**Review by Eric761 — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Pull request merged**